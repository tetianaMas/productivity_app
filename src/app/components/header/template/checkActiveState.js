/**
 * @description manages page hash and active css class according to it
 * @exports checkActiveState
 * @param {string} btnValue
 * @return {string} name of css class which represents active state of navigation button
 */
export default function (btnValue) {
  const pageHash = `${window.location.hash}/`.match(/^#\/(.*?)\//);

  if (pageHash && pageHash[1] === btnValue) {
    return 'navigation__btn--active';
  }
}
