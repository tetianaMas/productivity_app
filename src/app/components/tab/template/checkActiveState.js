/**
 * @description manages tabs for reports
 * @exports checkActiveState
 * @param {boolean} period if reports by time should be shown
 * @param {string} value current tab value
 * @return {string} active css class
 */
export default function (period, value) {
  if (period) {
    const regex = /(^#\/reports\/)(.*)\//gm;
    const match = regex.exec(`${window.location.hash}`);
    let time;

    if (match) {
      time = match[2];
    }

    if (time === value) {
      return 'tab__btn--active';
    }
  } else {
    const regex = /(^#\/reports\/)(.*)\/(.*)$/gm;
    const match = regex.exec(`${window.location.hash}`);
    let category;

    if (match) {
      category = match[3];
    }

    if (category === value) {
      return 'tab__btn--active';
    }
  }
}
