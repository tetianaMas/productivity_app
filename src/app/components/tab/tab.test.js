/* eslint-disable no-undef */
import tabs from './template/tab.hbs';
import checkActiveState from './template/checkActiveState';

describe('tabs rendering', () => {
  test('should render tab', () => {
    const context = {
      firstVal: 'pomodoros',
      secondVal: 'categories',
      activeTab: true,
      firstValAttr: 'data-tab="pomodoros"',
      secondValAttr: 'data-tab="categories"',
    };
    const tree = JSON.stringify(tabs(context));
    expect(tree).toMatchSnapshot();
  });
});

describe('checkActiveState', () => {
  let windowSpy;

  beforeEach(() => {
    windowSpy = jest.spyOn(window, 'window', 'get');
    windowSpy.mockImplementation(() => ({
      location: {
        hash: '#/reports/day/tasks',
      },
    }));
  });

  afterEach(() => {
    windowSpy.mockRestore();
  });

  test('should return active css class', () => {
    const result = checkActiveState('day', 'day');
    expect(result).toBe('tab__btn--active');
  });

  test('should return active css class', () => {
    const result = checkActiveState('', 'tasks');
    expect(result).toBe('tab__btn--active');
  });

  test('should return falsy result', () => {
    const result = checkActiveState('', '');
    expect(result).toBeFalsy();
  });
});
