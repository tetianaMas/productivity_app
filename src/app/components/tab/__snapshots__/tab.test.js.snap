// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabs rendering should render tab 1`] = `"\\"<div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n\\\\r\\\\n  <div class=\\\\\\"tab tab--right \\\\\\">\\\\r\\\\n      <button class=\\\\\\"tab__btn js-tablink first-main-tab tab__btn--active \\\\\\" data-tab=\\\\\\"pomodoros\\\\\\">pomodoros</button>\\\\r\\\\n      <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" data-tab=\\\\\\"categories\\\\\\">categories</button>\\\\r\\\\n  </div>\\\\r\\\\n</div>\\\\r\\\\n\\""`;
